### DISCUSSION TITLE
Source code for GPU features?

### DISCUSSION THREAD

**Discussion by HaydnVH** [NVIDIA-Omniverse/PhysX#22] [Q&A] [answered]
Currently, it seems that the GPU features of PhysX 5 are only available as closed-source binaries that require CUDA.  This makes these features a non-starter for many developers, who may seek out alternative solutions (AMD's FEMFX, for example) which will result in a fragmented system difficult to produce content for, or even choose a different physics library altogether.

The best case scenario (in my opinion), is what Flow seems to already be doing; full open source for a Vulkan/D3D12 version.  If this isn't practical, then releasing the source code for the existing CUDA implementations would allow the community to create their own Vulkan/D3D12 ports.

Will we ever see something like this in the future?

**Comment by guywmartin** [comment, marked as answer]
Thanks for the feedback @HaydnVH.  At this time, the decision was made to pursue an open core model where we keep the GPU features under a commercial license but provide the binaries free of charge.

**Reply by HaydnVH** [reply]
That's unfortunate.  Is there a chance this decision might change in the future?
